### PR TITLE
Fix idea plugin convention mappings (#8749)

### LINKED ISSUE #8749: For idea plugin, `.add`ing an excluded folder does not work, only reassignment works
### Expected Behavior

For

```
idea {
    module {
        excludeDirs.add(file("testData"))
    }
}
```

I expect that `testData` will be an excluded folder in IDE.

### Current Behavior

`testData` is not an excluded folder. However, if I use more verbose `excludeDirs = excludeDirs + file("testData")`, `testData` is indeed excluded. 

### Context

This seems like a bug either with DSL or `IdeaModule`. Note that `generatedSourceDirs.add` works fine. Here's the project where I am facing the issue: https://github.com/intellij-rust/intellij-rust/blob/c150f089e59a99e61ddbd16a178d096b17116010/build.gradle.kts

### PR DESCRIPTION
**Pull request by Vampire**
Fixes #8749

### Contributor Checklist
- [x] [Review Contribution Guidelines](https://github.com/gradle/gradle/blob/master/CONTRIBUTING.md)
- [x] Make sure that all commits are [signed off](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---signoff) to indicate that you agree to the terms of [Developer Certificate of Origin](https://developercertificate.org/).
- [x] Make sure all contributed code can be distributed under the terms of the [Apache License 2.0](https://github.com/gradle/gradle/blob/master/LICENSE), e.g. the code was written by yourself or the original code is licensed under [a license compatible to Apache License 2.0](https://apache.org/legal/resolved.html).
- [x] Check ["Allow edit from maintainers" option](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) in pull request so that additional changes can be pushed by Gradle team
- n/a Provide integration tests (under `<subproject>/src/integTest`) to verify changes from a user perspective
- [x] Provide unit tests (under `<subproject>/src/test`) to verify logic
- n/a Update User Guide, DSL Reference, and Javadoc for public-facing changes
- [x] Ensure that tests pass sanity check: `./gradlew sanityCheck`
- [x] Ensure that tests pass locally: `./gradlew <changed-subproject>:quickTest`

### Gradle Core Team Checklist
- [x] Verify design and implementation 
- [x] Verify test coverage and CI build status
- [x] Verify documentation
- [ ] Recognize contributor in release notes

### CODE REVIEW

File: subprojects/ide/src/main/java/org/gradle/plugins/ide/idea/IdeaPlugin.java [hunk 1]
```diff
@@ -302,11 +302,11 @@ public File call() {
         });
 
         Set<File> excludeDirs = Sets.newLinkedHashSet();
+        excludeDirs.add(project.file(".gradle"));
```
Comment (oehme) [inline subprojects/ide/src/main/java/org/gradle/plugins/ide/idea/IdeaPlugin.java:305]: This changes behavior, evaluating the build directory early.

Comment (Vampire) [reply subprojects/ide/src/main/java/org/gradle/plugins/ide/idea/IdeaPlugin.java]: D'oh your right of course, there is even a test for that.
Bad me first create the PR and then kicked off the tests.
Should be better now hopefully, thanks.

**Comment by big-guy** [comment]
@bot-gradle test this

**Review by ljacomet — APPROVED** [review]

**Comment by ljacomet** [comment]
@bot-gradle test and merge

**Pull request merged**